### PR TITLE
Add sidecar file size limitations

### LINKED ISSUE #3700: Add limit to sidecar file sizes
### Problem

When doing performance tests on the sidecar files produced by record stream v6, there were some very large files produced (more than 1GB). This can lead to problems with memory on the mirror node side.

### Solution

Introduce a limit to sidecar records file sizes and create multiple files when needed.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by dimitar-dinev**
**Description**:

Instead of creating one gigantic [record stream V6 sidecar file](https://github.com/hashgraph/hedera-services/pull/3625) in certain scenarios ( > `1GB`), have a size limit and create multiple smaller sidecars, which can be digested easier by the mirror nodes.

Enforces a sidecar size limit in `RecordStreamFileWriter` defined by the property `hedera.recordStream.sidecarMaxSizeMb`. The limit is currently set to `256MB`. 

**Related issue(s)**:

Fixes #3700

### CODE REVIEW

**Review by stoqnkpL — DISMISSED** [review]
LGTM

File: hedera-node/src/main/java/com/hedera/services/stream/RecordStreamFileWriter.java [hunk 1]
```diff
@@ -427,7 +435,28 @@ private void consume(final RecordStreamObject object) {
 
         final var sidecars = object.getSidecars();
         if (!sidecars.isEmpty()) {
-            for (final var sidecar : sidecars) {
+            for (final var sidecarBuilder : sidecars) {
+                // build() and getSerializedSize() would have been called anyway by the proto
+                // library downstream, so we do not incur any performance loses calling them here
```
Comment (georg-getz) [inline hedera-node/src/main/java/com/hedera/services/stream/RecordStreamFileWriter.java:440]: double 's' when used as a noun - losses

**Review by IvanKavaldzhiev — APPROVED** [review]
LGTM

File: hedera-node/src/main/java/com/hedera/services/stream/RecordStreamFileWriter.java (hunk 1 above)
Comment (dimitar-dinev) [reply hedera-node/src/main/java/com/hedera/services/stream/RecordStreamFileWriter.java]: Fixed

**Review by tinker-michaelj — APPROVED** [review]
LGTM

**Review by tinker-michaelj — APPROVED** [review]
LGTM

**Pull request merged**